src/IdeaStatiCa.IntermediateModel/Extensions/SObjectExtension.cs

```diff
@@ -0,0 +1,254 @@
+﻿using IdeaStatiCa.IntermediateModel.IRModel;
+
+namespace IdeaStatiCa.IntermediateModel.Extensions
+{
+	static public class SObjectExtension
+	{
+		/// <summary>
+		/// Add Element Property
+		/// </summary>
+		/// <param name="sObject"></param>
+		/// <param name="property"></param>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static void AddElementProperty(this SObject sObject, ISIntermediate property)
+		{
+			if (!sObject.TryAddElementProperty(property))
+			{
+				throw new InvalidOperationException($"Object {sObject.TypeName} already contains property {property.GetElementName()}");
+			}
+		}
+
+		/// <summary>
+		/// Try Add Element Property
+		/// </summary>
+		/// <param name="sObject"></param>
+		/// <param name="property"></param>
+		/// <returns></returns>
+		/// <exception cref="InvalidOperationException"></exception>
+		public static bool TryAddElementProperty(this SObject sObject, ISIntermediate property)
+		{
+			if (sObject.Properties.ContainsKey(property.GetElementName()))
+			{
+				return false;
+				throw new InvalidOperationException($"Object {sObject.TypeName} already contains property {property.GetElementName()}");
+			}
+			else
+			{
+				sObject.Properties.Add(property.GetElementName(), property);
+			}
+			return true;
+		}
+
+		/// <summary>
+		/// Get Elements
+		/// </summary>
+		/// <param name="sObject"></param>
+		/// <param name="filter"></param>
+		/// <returns></returns>
+		static public IEnumerable<ISIntermediate> GetElements(this SObject sObject, Queue<string> filter)
+		{
+
+			if (filter == null || filter.Count == 0)
+			{
+				return new List<ISIntermediate>();
+			}
+
+			if (filter == null)
```

Comment:
this condition would never be executed, same condition is above

Reply:
removed